misk-hibernate/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -68,6 +68,28 @@ internal class ReflectionQuery<T : DbEntity<T>>(
     disabledChecks += check
   }
 
+  override fun <Q: Query<*>> clone(): Q {
```

Comment:
This is a scary hack but it works and I couldn't think of a better way of doing thiis

Reply:
Mind adding a simple regression test?

Reply:
Yeah good idea, added a test